luaclang.c

```diff
@@ -329,14 +329,19 @@ static int cursor_gettypdef_underlying(lua_State *L)
 enum CXChildVisitResult visitor_function(CXCursor cursor, CXCursor parent, CXClientData client_data)
 {
         lua_State *L = (lua_State*) client_data;
+        int nargs =  lua_gettop(L);
         lua_pushvalue(L, 1);    
         CXCursor *cur;
         new_object(L, cur, CURSOR_METATABLE);           
         *cur = cursor;
         CXCursor *par;
         new_object(L, par, CURSOR_METATABLE);
         *par = parent;
-        if (lua_pcall(L, 2, 1, 0) != 0) {
+        lua_checkstack(L, nargs-1);
```

Comment:
You should put this before the first `new_object`; change the number of slots require accordingly. Also, you should check the output of the function and trigger a Lua error if it fails. Check the docs:
https://www.lua.org/manual/5.3/manual.html#lua_checkstack